.config/gitpod/Dockerfile

```diff
@@ -38,23 +36,37 @@ RUN PYTHON_VERSION=$(cut -d '=' -f 2 /tmp/python_version.config) && \
         vim \
         wget \
     && python3 -m pip install --no-cache-dir --upgrade pip \
-    && python3 -m pip install --no-cache-dir virtualenv \
-    # Install Docker
-    && curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg \
+    && python3 -m pip install --no-cache-dir virtualenv
+
+# Install Docker
+FROM builder as docker-installer
+
+RUN curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg \
     && echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu \
     $(lsb_release -cs) stable" | tee /etc/apt/sources.list.d/docker.list > /dev/null \
     && apt-get update \
-    && apt-get install -y --no-install-recommends docker-ce docker-ce-cli containerd.io \
-    # Install npm and ungit using the versions specified in the variables
-    && curl -fsSL https://deb.nodesource.com/setup_20.x |bash - \
+    && apt-get install -y --no-install-recommends docker-ce docker-ce-cli containerd.io
+
+# Install npm and ungit
+FROM docker-installer as npm-installer
+
+RUN curl -fsSL https://deb.nodesource.com/setup_20.x | bash - \
     && apt-get install  -y --no-install-recommends nodejs \
     && npm install -g npm@${NPM_VERSION} \
     && npm install -g \
-        ungit@${UNGIT_VERSION} \
-    # Clean
-    && apt-get clean && rm -rf /var/lib/apt/lists/* tmp/*
+        ungit@${UNGIT_VERSION}
+
+# Cleanup
+FROM npm-installer as cleanup
+
+RUN apt-get clean && rm -rf /var/lib/apt/lists/* tmp/* 
+
+# Final stage: Setup final image and user
+FROM ubuntu:jammy
+
+COPY --from=cleanup / /
 
 # Create the gitpod user. UID must be 33333.
 RUN useradd -l -u 33333 -G sudo -md /home/gitpod -s /bin/bash -p gitpod gitpod
 
-USER gitpod
+USER gitpod
```

Comment:
There is a missing newline (or line break) at the end of the file.

This is often mentioned because the POSIX standard defines a line as a sequence of zero or more non-newline characters, followed by a newline character. Therefore, by convention, many tools (including git) expect a file to end with a newline. If this end line is missing, some tools may misinterpret the last line of the file, or display warnings that the file does not end with a newline.

Reply:
@ThomasSanson 

I have just made the modification in accordance with the POSIX standard.

You can find the modifications in this commit: c88a85f8dbfef4874bbe07375307892dc3d75ae3

Reply:
@AurelienToitot 

It's peculiar, in your commit there doesn't appear to be an added empty line at the end of the file.

Moreover, when I look at your branch, it's the same, no empty line at the end of the file.